Automattic/wp-calypso
# Email: Go to inbox button should open in a new tab

Closes https://github.com/Automattic/wp-calypso/issues/95863#issuecomment-2542223088

## Proposed Changes

This PR ensures that `Go to inbox` button opens in a new link when the user purchase a new professional email and is brought to thank you page:

![Screenshot 2024-12-13 at 3 56 47 PM](https://github.com/user-attachments/assets/2ad4a92f-a22a-4d13-a55f-cf78a641c3e9)

## Testing Instructions



* Navigate to the Calypso Live link generated below
* Navigate to a screen to add a new domain `/domains/add/{yourWPcomSite}`
* Select a domain to purchase 
* On the next screen, select `Professional Email`, fill in the form and password and click on `Purchase`
* Proceed to checkout and complete the purchase
* Once you are presented with the thank you page, you will see the `Go to inbox` link
* Click on the link and confirm that it opens in a new tab

## Pre-merge Checklist
